viadot/tasks/hubspot.py

```diff
@@ -0,0 +1,61 @@
+import json
+import pandas as pd
+
+from typing import List, Dict, Any
+from prefect import Task
+from prefect.tasks.secrets import PrefectSecret
+from prefect.utilities import logging
+from viadot.sources import Hubspot
+
+logger = logging.get_logger()
+
+
+class HubspotToDF(Task):
+    def __init__(
+        self,
+        *args,
+        **kwargs,
+    ):
+
+        super().__init__(
+            name="hubspot_to_df",
+            *args,
+            **kwargs,
+        )
+
+    def __call__(self):
+        """Download Hubspot data to a DF"""
+        super().__call__(self)
+
+    def to_df(self, result: list = None):
+        return pd.json_normalize(result)
+
+    def run(
+        self,
+        endpoint: str,
+        properties: List[Any] = None,
+        filters: Dict[str, Any] = {},
+        nrows: int = 1000,
+    ) -> pd.DataFrame:
+
+        hubspot = Hubspot()
```

Comment:
You are not passing `credentials` and any other argument to the source class. Is it correct?

Reply:
yes :D

Reply:
All of the functions that are based on connection with the source are in this file - or will be ;)